backend/test/integration/openai.test.ts

```diff
@@ -259,27 +231,21 @@ describe('OpenAI Integration Tests', () => {
 			initChatHistory,
 			defences,
 			chatModel,
-			message,
-			true,
-			sentEmails
+			message
 		);
 
 		const { chatResponse, chatHistory } = reply;
 
 		expect(reply).toBeDefined();
 		expect(chatResponse.completion?.content).toBe('Hi');
 		// check the chat history has been updated
-		expect(chatHistory.length).toBe(4);
+		expect(chatHistory.length).toBe(2);
 		// system role is removed from the start of the chat history
 		// rest of the chat history is in order
 		expect(chatHistory[0].completion?.role).toBe('user');
 		expect(chatHistory[0].completion?.content).toBe("I'm a user");
 		expect(chatHistory[1].completion?.role).toBe('assistant');
 		expect(chatHistory[1].completion?.content).toBe("I'm an assistant");
-		expect(chatHistory[2].completion?.role).toBe('user');
-		expect(chatHistory[2].completion?.content).toBe('Hello');
-		expect(chatHistory[3].completion?.role).toBe('assistant');
-		expect(chatHistory[3].completion?.content).toBe('Hi');
```

Comment:
Why have these been removed? How have we changed the implementation?

Reply:
so openai.ts is no longer in charge of adding user or bot messages to the chat history, this is now done in the chat controller.
The only messages that openai.ts adds to the chat history are function calls. Even this I want to move out of openai.ts, as mentioned in the PR description, but I don't think I have time to do so unfortunately.